### PR TITLE
Add a flag to force headless rendering mode

### PR DESCRIPTION
**Pull request by ahcorde**
🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

# 🎉 New feature

## Summary
This PR depends on https://github.com/ignitionrobotics/ign-rendering/pull/272. This will allow to use the flag `--headless` to force ignition-rendering to use the headless mode if it's available.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by chapulina** [comment]
I think the word "headless" can be confusing to Gazebo users. We often use that word to refer to a server-only simulation, without the GUI process.

Maybe `headless-rendering`?

**Comment by ahcorde** [comment]
@mjcarroll @chapulina open for review

File: src/cmd/cmdgazebo.rb.in [hunk 1]
```diff
@@ -73,6 +73,8 @@ COMMANDS = { 'gazebo' =>
   "                               enable console logging to a console.log          \n"\
   "                               file in the specified path.                      \n"\
   "\n"\
+  "  --headless-rendering         Run in rendering in headless mode                \n"\
```
Comment (chapulina) [inline src/cmd/cmdgazebo.rb.in:76]:
```suggestion
  "  --headless-rendering         Run rendering in headless mode                   \n"\
```

File: src/ign.hh [hunk 2]
```diff
@@ -62,7 +62,7 @@ extern "C" int runServer(const char *_sdfString,
     int _logCompress, const char *_playback,
     const char *_physicsEngine, const char *_renderEngineServer,
     const char *_renderEngineGui, const char *_file,
-    const char *_recordTopics);
+    const char *_recordTopics, int _headless);
```
Comment (chapulina) [inline src/ign.hh:65]: Missing doxygen

**Review by chapulina — APPROVED** [review]
I just have some minor comments, but the changes LGTM. I haven't tried actually running it headless with the upstream PR though.

I think it would be best to merge this only after https://github.com/ignitionrobotics/ign-rendering/pull/272 gets in, so the functionality is working.

**Review by chapulina — APPROVED** [review]
I just tried the flag and it works for me, but I haven't tried it in a container without access to the display.

**Pull request merged**